homeassistant/components/teslemetry/__init__.py

```diff
@@ -130,5 +128,5 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload Teslemetry Config."""
     if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
-        hass.data[DOMAIN].pop(entry.entry_id)
+        del entry.runtime_data
```

Comment:
This _feels_ weird, and I feel like Home Assistant should do this (it currently doesn't).